src/commands/watch.ts

```diff
@@ -80,6 +80,14 @@ export function _makeCurrentMigrationRunner(
 
             // 4: if different
             if (!migrationsAreEquivalent) {
+              if (parsedSettings.beforeCurrent) {
+                await executeActions(
+                  parsedSettings,
+                  shadow,
+                  parsedSettings.beforeCurrent,
+                );
+              }
```

Comment:
I think this is the most sensible place for this currently.

Reply:
Yes, and it is was what I needed, but added the others for completeness. Even it is yet v0, I don't think we want to add them and remove later. So let me know if you want, I will remove the other hooks.